.github/workflows/test.yml

```diff
@@ -68,19 +68,41 @@ jobs:
           echo "Public IP Address of the runner:"
           curl -s https://api.ipify.org
         shell: bash
+      - name: Download passed-marker artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: passed-marker-${{ github.job }}-${{ github.sha }}-${{ matrix.file_system }}-${{ matrix.tpm }}
+          path: ./marker
+        continue-on-error: true
+      - name: Check if already passed
+        id: test_passed
+        run: |
+          if [[ -f ./marker/passed ]]; then
+            echo "already_passed=true" >> "$GITHUB_OUTPUT"
+            rm -f ./marker/passed
+          else
+            echo "already_passed=false" >> "$GITHUB_OUTPUT"
+          fi
+      - name: Early exit if already passed
+        if: steps.test_passed.outputs.already_passed == 'true'
+        run: |
+          echo "Test for $GITHUB_SHA ${{ github.job }} (${{ matrix.file_system }}, tpm=${{ matrix.tpm }}) already passed. Skipping."
+          exit 0
       - name: Dockerhub Login
-        if: ${{ github.event.repository.full_name }} == 'lf-edge/eve'
+        if: ${{ github.event.repository.full_name == 'lf-edge/eve' && steps.test_passed.outputs.already_passed != 'true' }}
```

Comment:
why comparison if we have early exit?

Reply:
@uncleDecart, ah, the problem that exit 0 only stops the step, but not the entire job...

Reply:
Removed the unnecessary early exits.